client-assertion-validation/src/main/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidator.scala

```diff
@@ -62,43 +62,50 @@ final class NimbusClientAssertionValidator(expectedAudience: Set[String]) extend
     clientId: Option[UUID],
     clientAssertion: String
   ): Either[NonEmptyList[ValidationFailure], ClientAssertion] =
-    (
-      validateStandardClaims(jwt),
-      getOrFail(jwt.getHeader.getKeyID, KidNotFound),
-      algorithm(jwt.getHeader.getAlgorithm.getName),
-      subjectClaim(clientId, jwt.getJWTClaimsSet.getSubject),
-      purposeIdClaim(jwt.getJWTClaimsSet.getStringClaim(PURPOSE_ID_CLAIM)),
-      getOrFail(jwt.getJWTClaimsSet.getJWTID, JtiNotFound),
-      getOrFail(jwt.getJWTClaimsSet.getIssueTime.getTime, IssuedAtNotFound),
-      getOrFail(jwt.getJWTClaimsSet.getIssuer, IssuerNotFound),
-      audience(jwt.getJWTClaimsSet.getAudience.asScala.toSet, expectedAudience),
-      getOrFail(jwt.getJWTClaimsSet.getExpirationTime.getTime, ExpirationNotFound),
-      digestClaim(jwt.getJWTClaimsSet)
-    ).tupled.map { case (_, kid, algorithm, subject, purposeId, jti, issuedAt, issuer, audience, expiration, digest) =>
-      ClientAssertion(
-        kid = kid,
-        alg = algorithm,
-        sub = subject,
-        purposeId = purposeId,
-        jti = jti,
-        iat = issuedAt,
-        iss = issuer,
-        aud = audience,
-        exp = expiration,
-        digest = digest,
-        raw = clientAssertion
-      )
-    }.toEither
+    for {
+      claimSet        <- validateStandardClaims(jwt)
+      clientAssertion <- (
+        getOrFail(jwt.getHeader.getKeyID, KidNotFound),
+        algorithm(jwt.getHeader.getAlgorithm.getName),
+        subjectClaim(clientId, claimSet.getSubject),
+        purposeIdClaim(claimSet.getStringClaim(PURPOSE_ID_CLAIM)),
+        getOrFail(claimSet.getJWTID, JtiNotFound),
+        getOrFail(claimSet.getIssueTime.getTime, IssuedAtNotFound),
+        getOrFail(claimSet.getIssuer, IssuerNotFound),
+        audience(claimSet.getAudience.asScala.toSet, expectedAudience),
+        getOrFail(claimSet.getExpirationTime.getTime, ExpirationNotFound),
+        digestClaim(claimSet)
+      ).tupled.map { case (kid, algorithm, subject, purposeId, jti, issuedAt, issuer, audience, expiration, digest) =>
```

Comment:
isn't `.tupled.map` equivalent to `.mapN` ?

Reply:
unfortunately not, because the constructor takes one more parameter that is not coming from the tuple